CRM/Utils/Array.php

```diff
@@ -1058,16 +1058,20 @@ public static function formatArrayKeys(&$array) {
     if (!is_array($array)) {
       return;
     }
-    $keys = array_keys($array, 1);
-    if (count($keys) > 1 ||
-      (count($keys) == 1 &&
-        (current($keys) > 1 ||
-          is_string(current($keys)) ||
-          (current($keys) == 1 && $array[1] == 1) // handle (0 => 4), (1 => 1)
-        )
-      )
-    ) {
-      $array = $keys;
+
+    $hasSequentialKeysStartingWithZero = array_keys($array) === range(0, count($array) - 1);
```

Comment:
this can be simplified to : 

```
private static function doesRequireFormat($array) {
    $hasSequentialKeysStartingWithZero = array_keys($array) === range(0, count($array) - 1);
    $hasAllValuesAsNumberOne = array_unique(array_values($array)) === ['1'];
    $hasMoreThanOneElement = count($array) > 1;
    return !$hasSequentialKeysStartingWithZero || ($hasMoreThanOneElement && $hasAllValuesAsNumberOne);
}

```

```
if (self::doesRequireFormat($array)) {
$tempArr = [];
 ..etc
```

Reply:
Fair enough. However, I will need to rewrite this function hence failing tests. So I will take this suggestion into consideration if I have that much code.